ablateLibrary/domain/subDomain.hpp

```diff
@@ -0,0 +1,56 @@
+#ifndef ABLATELIBRARY_SUBDOMAIN_HPP
+#define ABLATELIBRARY_SUBDOMAIN_HPP
+
+#include <petsc.h>
+#include <map>
+#include <memory>
+#include <string>
+#include "domain.hpp"
+#include "fieldDescriptor.hpp"
+
+namespace ablate::domain {
+
+class SubDomain {
+   private:
+    // const pointer to the parent domain/dm
+    const std::weak_ptr<Domain> domain;
+
+    // The label used to describe this subDomain
+    DMLabel label;
```

Comment:
I think you might want (label, value) or even (label, {values}) to specify the subdomain

Reply:
Yep, that really needs to be fixed. I guess I should fix it now, before anyone else builds code on it.

Reply:
@knepley Thanks!  I've added a new struct for a region, that the solvers specify.  This includes a label name and values.  The subDomain now uses a label and values and defaults to "1".